admin/client/src/state/schema/SchemaReducer.js

```diff
@@ -10,6 +11,16 @@ export default function schemaReducer(state = initialState, action = null) {
       return deepFreeze(Object.assign({}, state, { [action.payload.id]: action.payload }));
     }
 
+    case ACTION_TYPES.CLEAR_MESSAGE: {
```

Comment:
The other approach here would be to have a value in the state that is the incorrect field value. The error message will only be shown if the field is still set to that value. This means that you don't need a state change to clear the error message, just a re-render:

```
...
{
  message: {
    'value': 'Not a valid email address',
    'for': 'sam@silverstripe,com',
  }
...

```

It will need a re-render to trigger—presumably `onblur` rather than `onchange`